keel-web/src/main/kotlin/com/netflix/spinnaker/config/SlackRequestSecurityConfiguration.kt

```diff
@@ -0,0 +1,29 @@
+package com.netflix.spinnaker.config
+
+import org.springframework.context.annotation.Configuration
+import org.springframework.core.annotation.Order
+import org.springframework.security.config.annotation.web.builders.HttpSecurity
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
+
+/**
+ * Disables several security-related filters for Slack callback requests, which are application/x-www-form-urlencoded
+ * and cause the filters to read the request body while reading parameters, which in turn causes the input stream
+ * to be closed before the request gets to the callback controller, appearing as an empty request body.
+ */
+@Configuration
+@Order(1)
+@EnableWebSecurity
+class SlackRequestSecurityConfiguration : WebSecurityConfigurerAdapter() {
+
+  override fun configure(http: HttpSecurity) {
+    http
+      .antMatcher("/slack/notifications/callbacks")
+      .authorizeRequests()
+      .anyRequest()
+      .permitAll()
+      .and()
+      .csrf()
+      .ignoringAntMatchers("/slack/notifications/callbacks")
+  }
```

Comment:
I THINK this is problematic... as that'd allow non authenticated access to that URL which... would allow any slack notification.  I'm asking around... b/c I may have the order wrong atm...

Reply:
Good catch, I ended up reverting [this change](https://github.com/luispollo/keel/commit/b56ccbd5b0a91e30f6a24dfbd57609e8024c7cf5#diff-48ff33f739abaec5bfa33e61576ba2997c57035cc5f69aabdd283b98e9eb2f45) since it relied on some security workarounds. That commit could be revisited if we ever decide to port it over